src/controllers/activegate/reconciler/statefulset/statefulset.go

```diff
@@ -525,3 +525,7 @@ func isCustomPropertiesNilOrEmpty(customProperties *dynatracev1beta1.DynaKubeVal
 		(customProperties.Value == "" &&
 			customProperties.ValueFrom == "")
 }
+
+func needsProxy(stsProperties *statefulSetProperties) bool {
+	return !stsProperties.DynaKube.FeatureActiveGateIgnoreProxy() && stsProperties.HasProxy()
```

Comment:
IIRC, `stsProperties` embeds the DynaKube, so the call can be simplified to 
```suggestion
	return !stsProperties.FeatureActiveGateIgnoreProxy() && stsProperties.HasProxy()
```

Reply:
could then be further refactored into a property of DynaKube called, e.g., `ActiveGateNeedsProxy()`.
I.e.,
```
// In properties.go
func (dynakube *DynaKube) ActiveGateNeedsProxy() bool {
  return !stsProperties.FeatureActiveGateIgnoreProxy() && stsProperties.HasProxy()
}

// In statefulset.go
if stsProperties.ActiveGateNeedsProxy() {
```